iceoryx2/src/service/builder/mod.rs

```diff
@@ -98,17 +98,28 @@ impl std::error::Error for ReadStaticStorageFailure {}
 #[derive(Debug)]
 pub struct Builder<S: Service> {
     name: ServiceName,
+    properties: ServiceProperties,
     _phantom_s: PhantomData<S>,
 }
 
 impl<S: Service> Builder<S> {
     pub(crate) fn new(name: &ServiceName) -> Self {
         Self {
-            name: *name,
+            name: name.clone(),
+            properties: ServiceProperties::new(),
             _phantom_s: PhantomData,
         }
     }
 
+    /// Defines a property requirement. If a new [`Service`] is created all properties will be
+    /// added. If an existing [`Service`] is opened those properties are interpreted as
+    /// requirements that the service has to satisfy. If a property does not match or does not
+    /// exist the open process will fail.
```

Comment:
When you create a service, you can set its properties; if you open a service, the defined properties are interpreted as property requirements.

When opening a service:
1. If you do not specify any property, it is interpreted as having no property requirement.
2. If you specify a property and the key with the same value is defined in the service, you can open the service.
3. Open fails if you specify a property where the key is not listed in the service or the key value is not part of the property.

Reply:
It would also be nice to have something like a wildcard value, i.e. requiring only the key being present but reading the value later on once the service is opened. With this a global static config could be delivered just as a service.